functions/documents.ts

```diff
@@ -1,25 +1,465 @@
+ computed: Pick<DocumentComputed, 'isDraft' | 'velocity' | 'score' | 'charCount'>,
+ now = Date.now()
+) {
+ const { isDraft, velocity, score, charCount } = computed
+ if (velocity === 0) return
+
+ if (!isDraft && velocity < 0) {
+  const delta = score - charCount
+  if (delta <= 0) return
+  const hours = delta / Math.abs(velocity)
+  const unpublishesAt = new Date(now + hours * ONE_HOUR_MS)
+  return `Unpublishes on ${unpublishesAt.toLocaleString()}`
+ }
+
+ if (isDraft && velocity > 0) {
+  const delta = charCount - score
+  if (delta <= 0) return
+  const hours = delta / velocity
+  const publishesAt = new Date(now + hours * ONE_HOUR_MS)
+  return `Publishes on ${publishesAt.toLocaleString()}`
+ }
+
+ if (isDraft && velocity < 0) {
+  if (score <= 0) {
+   const expiresAt = new Date(now)
+   return `Draft expires on ${expiresAt.toLocaleString()}`
+  }
+  const hours = score / Math.abs(velocity)
+  const expiresAt = new Date(now + hours * ONE_HOUR_MS)
+  return `Draft expires on ${expiresAt.toLocaleString()}`
+ }
+}
+
+function toComputed(
+ doc: StoredDocument,
+ includeBody: boolean,
+ now = Date.now()
+): DocumentComputed {
+ const score = calculateScore(doc.vote, now)
+ const velocity = doc.vote.velocity
+ const charCount = doc.body.length
+ const isDraft = score < charCount
+
+ const tagScores: Record<string, number> = {}
+ const tagVelocities: Record<string, number> = {}
+ for (const tag of doc.tags) {
+  const vote = doc.tagVotes[tag]
+  const effectiveVote = vote ?? {
+   position: 0,
+   velocity: 0,
+   timestamp: doc.created,
+  }
+  tagScores[tag] = calculateScore(effectiveVote, now)
+  tagVelocities[tag] = effectiveVote.velocity
+ }
+
+ const computed: DocumentComputed = {
+  id: doc.id,
+  title: doc.title,
+  tags: doc.tags,
+  created: doc.created,
+  modified: doc.modified,
+  rev: doc.rev,
+  score,
+  velocity,
+  charCount,
+  isDraft,
+  statusMessage: computeStatusMessage({ isDraft, velocity, score, charCount }, now),
+ }
+
+ // In list mode, expose only the top tags (net score > 1) per spec.
+ if (!includeBody) {
+  const topTags = doc.tags
+   .map((t) => {
+    const vote = doc.tagVotes[t] ?? {
+     position: 0,
+     velocity: 0,
+     timestamp: doc.created,
+    }
+    return { tag: t, score: calculateScore(vote, now) }
+   })
+   .filter((x) => x.score > 1)
+   .sort((a, b) => b.score - a.score)
+   .slice(0, 3)
+   .map((x) => x.tag)
+  computed.topTags = topTags
+ }
+ if (includeBody) {
+  computed.body = doc.body
+  computed.tagScores = tagScores
+  computed.tagVelocities = tagVelocities
+ }
+ return computed
+}
+
+function json(data: unknown, status = 200) {
+ return new Response(JSON.stringify(data), {
+  status,
+  headers: {
+   'Content-Type': 'application/json',
+  },
+ })
+}
+
+function errorText(message: string, status = 400) {
+ return new Response(message, { status })
+}
+
+async function listAllDocs(kv: any): Promise<StoredDocument[]> {
+ const { store } = await import('./lib/store.js')
```

Comment:
Move this to a static import at the top of the file, all operations require the `store`